dotCMS/src/main/java/com/dotcms/cluster/business/ServerAPIImpl.java

```diff
@@ -35,52 +44,111 @@ public void saveServer(Server server) throws DotDataException {
 	public Server getServer(String serverId) {
 		return serverFactory.getServer(serverId);
 	}
+	@Override
+    public Server getOrCreateServer(final String serverId) throws DotDataException {
+	    Server tryServer = getServer(serverId);
+        if(tryServer==null || tryServer.getServerId() ==null)  {
+            LocalTransaction.wrap(() ->{
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 28 to at most 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1188)